components/widgets/badges/badge.tsx

```diff
@@ -1,32 +1,37 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React from 'react';
+import React, {memo, ReactNode, HTMLAttributes} from 'react';
 
 import './badge.scss';
 
 type Props = {
-    show: boolean;
-    children: React.ReactNode;
-    className: string;
+    show?: boolean;
+    children: ReactNode;
 };
 
-export default class Badge extends React.PureComponent<Props> {
-    public static defaultProps = {
-        show: true,
-        className: '',
-    };
+type Attrs = HTMLAttributes<HTMLDivElement>
 
-    public render() {
-        if (!this.props.show) {
-            return null;
-        }
-        return (
-            <div className='Badge'>
-                <div className={'Badge__box ' + this.props.className}>
-                    {this.props.children}
-                </div>
-            </div>
-        );
+const Badge = ({
+    show = true,
+    children,
+    className,
+    ...attrs
+}: Props & Attrs) => {
+    if (!show) {
+        return null;
     }
-}
+    return (
+        <div className='Badge'>
+            <div
+                {...attrs}
+                role={attrs.onClick ? 'button' : undefined}
+                className={'Badge__box ' + className}
+            >
+                {children}
+            </div>
```

Comment:
If we are going to make this optionally clickable, I'd prefer to use an actual button or link element instead of the role attribute unless it absolutely doesn't make sense or for legacy reasons just isn't currently possible.

Reply:
I agree, and I can switch it out; was trying to keep changes to other codepaths as minimal as possible for the time being, but I guess it wouldn't be that large of an impact.